xenmgr/XenMgr/Connect/Xl.hs

```diff
@@ -209,7 +209,7 @@ unpause uuid = do
 getXlProcess :: Uuid -> IO String
 getXlProcess uuid = do
     let conv_uuid = TT.replace "_" "-" (show uuid)
-    (ec,str_pid,_) <- readProcessWithExitCode "pgrep" ["-f", conv_uuid] ""
+    (ec,str_pid,_) <- readProcessWithExitCode "pgrep" ["-f", "xl[a-zA-Z0-9\\w:\\.\\?\\+\\-]*" ++ conv_uuid] ""
```

Comment:
Minimally, a more robust check could start with "^xl " to ensure the match begins with xl and have a space to separate the command from arguments.  xenmgr starts the xl process with ```createProcess (proc "xl" ["create", configPath uuid, "-p"])``` so the expression could just be ```"^xl create " ++ configPath uuid ++ " -p"```?

Since we'll signaling processes, I still think it would be better to track pids explicitly and ensure we are only signaling processes xenmgr started.

Reply:
You're right about the expression to look for, I don't need that match set.  Will update.

You mentioned before the difficulty of tracking pids due to xl forking.  Xenmgr can know the pid from the initial "xl create" call, but getting the forked pid would either require a pgrep in some form anyway, or a message from xl -> xenmgr telling it the pid changed, which is something I'm trying to avoid.  There's only two things I want going from xl -> xenmgr: errors on failure, and domain state changes (which is more indirectly sent through xenstore).

Reply:
Yes, but you can use "-F" to run in the foreground and skip daemonizing.  With that the pid wouldn't change from the initial exec.  I don't know if that would then require further changes to work properly like waiting on children.

I was just re-stating my preference.  With the pgrep improvements this will work.  A future improvement could be to change over to tracking children.